offload/plugins-nextgen/amdgpu/src/rtl.cpp

```diff
@@ -57,6 +57,12 @@
 #endif
 
 #if defined(__has_include)
+#if __has_include("hip/hip_version.h")
+#include "hip/hip_version.h"
+#else
+#include "hip_version.h"
+#endif
+
```

Comment:
I'm guessing that HSA didn't bump up a `HSA_AMD_INTERFACE_VERSION_MINOR` for this feature or something?

Reply:
Yes, you are right! They didn't bump up HSA_AMD_INTERFACE_VERSION_MINOR. Despite new features, the HSA versions shipped with (at least) ROCm 6.0.2 and ROCm 6.1 have the same HSA_AMD_INTERFACE_VERSION_MINOR.

Reply:
I suppose we need to be more vocal to the HSA developers that we really need the API to be versioned whenever a new interface is added. I really don't want us to depend on HIP here just for a version check that should be in HSA anyway.

Reply:
I talked to one of the HSA guys. HSA_AMD_INTERFACE_VERSION_MINOR will be bumped up and made available with the 6.2 release. An internal PR for this has already been created. So, I tend to ride it out. I will adjust the #if condition to check for HSA_AMD_INTERFACE_VERSION_MINOR >= 5.

Reply:
Okay, so we'll automatically pick this up once the ROCm installation is upgrades? How about in the `dlopen(libhsa-runtime64.so` case? I think we want an actual function to get the versions as well in those cases.

Reply:
Yes, if ROCm 6.2 is eventually available during the build process, the new implementation will be used. 

Regarding your second question. Are you referring to a method that determines the HSA version at runtime?